framework/include/meshgenerators/MeshDiagnosticsGenerator.h

```diff
@@ -32,9 +32,13 @@ class MeshDiagnosticsGenerator : public MeshGenerator
   /// Routine to check sideset orientation near subdomains
   void checkSidesetsOrientation(const std::unique_ptr<MeshBase> & mesh) const;
   //// Routine to check is mesh is fully covered in sidesets
-  void checkWaterTightSidesets(const std::unique_ptr<MeshBase> & mesh) const;
+  void checkWatertightSidesets(const std::unique_ptr<MeshBase> & mesh) const;
   //// Routine to check is mesh is fully covered in nodesets
   void checkWatertightNodesets(const std::unique_ptr<MeshBase> & mesh) const;
+  /// Helper function that finds the intersection of the _watertight_boundaries vector and the provided one
+  std::vector<boundary_id_type>
+  findBoundaryOverlap(std::vector<boundary_id_type>,
+                      std::vector<boundary_id_type> boundary_ids) const;
```

Comment:
```suggestion
  findBoundaryOverlap(const std::vector<boundary_id_type> & boundary_copy,
                      const std::vector<boundary_id_type> & boundary_ids) const;
```

Reply:
So in this function I do sort the one vector. If I do that then I don't think I can make it const?

Reply:
No, you can't.  You should put the requirement for the vector in a comment above the method, though, and add an assertion for it in the method.

Don't both vectors have to be sorted for a `set_intersection()`, though?

Reply:
Yes they do. The one vector (boundary_copy) is a copy of the const vector that contains the list of boundaries specified in the input file. It's sorted before it's inputted. The other is sorted inside the function. I feel like I had a good reason for sorting the one outside of the function and one inside but I can't remember why. If it comes to me I'll let you know,

Reply:
It's reasonable to require a vector to be sorted outside the function for performance reasons (you can then reuse the sorted property over multiple invocations with the same vector with no call to a sort inside the function, which is good since even a no-op sort is still O(N)).  It's a little weird to require this of one vector but not the other, but for a specialized little function like this and a use case where you only reuse one input that is the simplest thing to do.

Why not pass the unsorted vector by non-const reference, though?  If you have a non-const vector to hand it, and you're either not reusing or not relying on the ordering of that vector afterwards, which seems to be the case here, then it's faster to sort the input rather than copying the input and sorting that.